### ISSUE TITLE
Kernel size selection and training

### ISSUE THREAD

**Issue by ManojPrabhakar** [jacobkimmel/pytorch_convgru#3]
Hello,

I have a project in which I am using convgru. I have some doubts regarding the same:

1. I am using convgru for tracking. My input is a grayscale image and my output of autoencoder model is a grayscale map of dimension [4,1,128,160]. Since I am using a batch size of 4 what should be my kernal and hidden size? I am currently using input_size = 1 (grayscale image), hidden_sizes=[1,64,64,64], kernel_sizes=[3,3,3,3], n_layers=4. It's giving no errors but are these correct values or do I have to find them by trial and error?

2. I have output labels as well. How do I train this model?

Regards,
Manoj

**Comment by jacobkimmel** [comment]
> Since I am using a batch size of 4 what should be my kernal and hidden size?

It's difficult for me to provide input on what the hyperparameters should be for your task. My suggestion is to pick something sensible (like `64`) and then run a search to optimize these hyperparameters. 

>  I have output labels as well. How do I train this model?

I'm not sure what your task is, but generally you define a loss based on the model outputs and your desired output then use an optimization strategy to optimize it. Try looking through the PyTorch tutorials. 

https://pytorch.org/tutorials/
